### DISCUSSION TITLE
Email Auth: Sign Up Disabled (Invite Only) & Sign In Allowed

### DISCUSSION THREAD

**Discussion by ay-22** [supabase/supabase#4296] [Questions]
Hi,

I'm working on a website where I want to users can only be invited via the Supabase dashboard. I'm building out a frontend sign in page for email and password. I'm finding that I can successfully invite a user through the dashboard with email sign ups turned off in auth settings. The problem is I can't make any sign in auth requests with sign ups turned off.

My thought process to achieve that was:

1. Disable Sign Ups in the Auth Settings
2. Invite Users via email through the Supabase Dashboard
3. User receives email and clicks on link and creates a password
4. Users who have existing accounts can sign in with email/password

I was able to get to step #3. Step #4 I attempted to make this request:
```
const { error, data } = await supabase.auth.signIn({
      email: email,
      password: password
    })
```

The following error (400 status code) message occurred: "Email logins are disabled". This makes sense in theory by disabling sign ups it seems to be disabling all authentication.

Does anyone have any suggestions around solving for this use case? Ideally you can toggle on allowing sign ups/sign ins in the auth settings.  Let me know if more clarification is needed.

Thanks

**Comment by TomasHubelbauer** [comment]
This seems related: https://github.com/supabase/gotrue/issues/190

I would also like to prevent new signups without affecting new logins of existing users. My application is intended for a limited audience and having control over who is allowed to sign up or not is crucial as to not reveal non-public information.

Issue with more context supabase/gotrue#330 and a possible fix https://github.com/supabase/gotrue/pull/324

**Reply by shawnmclean** [reply]
Any pointers on how to do this with supabase?

I'm evaluating this, nextauthjs and clerk.

**Reply by JanRuettinger** [reply]
Have you found a solution?

**Reply by shawnmclean** [reply]
Rolled my own.

These things are opinionated wrappers around user experience related to authentication.

If you want control over the UX, more than just logging in, but things like invites, sign up flows, etc, grab your own OAuth/OpenId/JWT/Cookie low level wrappers and work from there.

These auth services will hold you back if you're pedantic about certain flows or UX.

**Comment by OusmaalK** [comment]
Hello here my code 
onPressed: () async {
                  try {
                    // Vérifier d'abord si l'utilisateur existe dans la table profile_user
                    final response = await supabase
                        .from('profile_user')
                        .select()
                        .eq('email', emailController.text)
                        .limit(1)
                        .execute();

                    // Si l'utilisateur existe, connectez-le
                    if (response.data != null &&
                        (response.data as List).isNotEmpty) {
                      final signInResponse =
                          await supabase.auth.signInWithPassword(
                        email: emailController.text,
                        password: passwordController.text,
                      );

                      final Session? session = signInResponse.session;
                      final User? user = signInResponse.user;

                      print("Utilisateur connecté avec succès");
                      _showWelcomeDialog();

                      // Naviguez vers une autre page (par exemple, "/home")
                      Navigator.pushReplacementNamed(context, "/home");
                    } else {
                      // Si l'utilisateur n'existe pas, affichez un message d'erreur
                      showSnackBar(
                          "Utilisateur non trouvé. Veuillez vous inscrire.");
                    }
                  } catch (error) {
                    print("Erreur de connexion : $error");
                    showSnackBar("Erreur de connexion : $error");
                  }
                },
                style: ElevatedButton.styleFrom(
                  backgroundColor: Colors.green,
                  padding: EdgeInsets.symmetric(vertical: 15),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(10),
                  ),
                ),
                child: Text(
                  "Connexion",
                  style: TextStyle(
                    fontSize: 18,
                    color: Colors.white,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ),
It appears that the provided text is in French and discusses checking if a user exists in the "profile_user" table and encountering an issue with the `.execute()` method in a query. Here's a translation in English:

"First, check if the user exists in the 'profile_user' table, and if there is a problem with the `.execute()` method in the query, please guide me towards a solution."

If you have a specific issue with the `.execute()` method or need further assistance, feel free to provide more details so that I can help you more effectively.

**Comment by avaid96** [comment]
For anyone looking still, now there's a way to do this from Project Settings
<img width="1261" alt="image" src="https://github.com/supabase/supabase/assets/9308841/7584eb2d-3cd0-4266-badf-98fb348e916e">

**Reply by Hronom** [reply]
I would like to have this option per provider

**Comment by Hronom** [comment]
Is it possible to add an option to enable sign-up per provider? 

So, for example, I need to allow users to sign up through social networks (google, facebook, etc), but disable signup through email and for email allow only sign-in. 
This will improve security, as we can create users from backend using a service key with email, but an anon key is not allowed to register users via email.
